src/main/java/domain/board/Direction.java

```diff
@@ -0,0 +1,62 @@
+package domain.board;
+
+import java.util.Arrays;
+
+public enum Direction {
+
+    UP(0, 1),
+    UPRIGHT(1, 1),
+    RIGHT(1, 0),
+    DOWNRIGHT(1, -1),
+    DOWN(0, -1),
+    DOWNLEFT(-1, -1),
+    LEFT(-1, 0),
+    UPLEFT(-1, 1);
+
+    private final int fileDirection;
+    private final int rankDirection;
+
+    Direction(int fileDirection, int rankDirection) {
+        this.fileDirection = fileDirection;
+        this.rankDirection = rankDirection;
+    }
+
+    public static Direction of(Position sourcePosition, Position targetPosition) {
+        int fileDifference = targetPosition.calculateFileDifference(sourcePosition);
+        int rankDifference = targetPosition.calculateRankDifference(sourcePosition);
+
+        return Arrays.stream(values())
+            .filter(direction -> direction.fileDirection == decideFileDirection(fileDifference)
+                && direction.rankDirection == decideRankDirection(rankDifference))
+            .findFirst()
+            .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 방향입니다."));
+    }
+
+    public static int decideFileDirection(int fileDifference) {
+        if (fileDifference > 0) {
+            return 1;
+        }
+        if (fileDifference < 0) {
+            return -1;
+        }
+        return 0;
+    }
```

Comment:
```suggestion
    public static int decideFileDirection(int fileDifference) {
        return Integer.compare(fileDifference, 0);
    }
```
자바에서 제공해주는 기능을 적극적으로 활용해보는건 어떨까요?

Reply:
Integer.compare로 바꾸는 것이 기존의 코드에 비해서 가독성이 떨어지는 것 같다고 생각합니다.

compare함수를 보고 의미를 바로 알아차리기 어렵다고 생각하는데,
피케이는 어떻게 생각하시나요?

Reply:
바로 알아차리기 어려운 것은 동의해요. 그런데 메서드에 커서를 올리면 설명이 나오기도 해서 저는 딱히 불편함을 못 느꼈던 것 같아요.
저는 `자바에서 제공해주는 기능을 굳이 다시 만들 필요가 있나?` 라는 생각으로 접근했었는데, 카피에게 이게 더 맞다면 그냥 유지해도 괜찮을 것 같네요 :)